### DISCUSSION TITLE
Mapped relationship annotation use case when model defined in different files

### DISCUSSION THREAD

**Discussion by wangxin688** [sqlalchemy/sqlalchemy#9223] [Usage Questions] [answered]
Maybe it's a silly question. 
if a lot models defined in different files. Use `Mapped['Object']` or `Mapped[List["Object"]]`will failed with undefined name if `Object` in another file.
sure I can import the Object from files it belongs to. if relationship is defined with `brakref`, OK, it's working fine, no circular import.
But if I defined relationship with `back_populates`, I need to import 'Object' to each other for Mapped, this action will cause circular import issue.
So, should I change back_populates way to backref for single direction import?
`User` to `Address` and  `UserD` to `Address`
file A:
```python
class User(Base):
    __tablename__ = "user"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str]
    firstname: Mapped[str] = mapped_column(String(50))
    lastname: Mapped[str] = mapped_column(String(50))
    fullname: Mapped[str] = column_property(firstname + " " + lastname)

    addresses: Mapped[List["Address"]] = relationship(back_populates="user")


class Address(Base):
    __tablename__ = "address"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("user.id"))
    userd_id: Mapped[int] = mapped_column(ForeignKey("userd.id"))
    email_address: Mapped[str]
    address_statistics: Mapped[Optional[str]] = mapped_column(Text, deferred=True)

    user: Mapped["User"] = relationship(back_populates="addresses")
    userd: Mapped["UserD"] = relationship(back_populates="addresses")
```

file B:
```python
class UserD(Base):
    __tablename__ = "userd"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str]
    firstname: Mapped[str] = mapped_column(String(50))
    lastname: Mapped[str] = mapped_column(String(50))
    fullname: Mapped[str] = column_property(firstname + " " + lastname)

    addresses: Mapped[List["Address"]] = relationship(back_populates="user")
```

**Comment by zzzeek** [comment, marked as answer]
you have to import both files at the point at which the mappers are [configured](https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.configure_mappers), but the order in which they are imported does not matter as long as the mappers are not being configured in that process.

**Reply by wangxin688** [reply]
Thanks a lot. It's not sqlalchemy issue. After rename import model, circular import was resolved. BTW, `Mapped` is excellent for type-hint.
